lib/oauth.js

```diff
@@ -28,9 +28,24 @@ class OAuth {
       .value();
   }
 
+  extractRedirectUri(req) {
+    return req.query['redirect_uri'];
+  }
+
   redirectURL(uuid, res) {
-    let base = _.get(this.link.chain.config, 'auth.redirectUrl');
+    const redirectDomains = _.get(this.link.chain.config, 'auth.redirectDomains');
```

Comment:
Can we default in the already-known Amazon domains so that everyone doesn't have to plug in the same configuration?